draft-webtransport-http2.md

```diff
@@ -145,47 +144,55 @@ application that accepts WebTransport sessions, which can be accessed via an
+by any intermediaries to correctly forward the stream to the destination
+endpoint.  The only frames allowed on WebTransport Streams are WTHEADERS,
+CONTINUATION, DATA and any negotiated extension frames.
 
 
-## The "webtransport" HTTP Upgrade Token
+## Negotiation
 
-Use of the extended CONNECT method extension defined in {{!RFC8441}}
-requires the SETTINGS_ENABLE_CONNECT_PROTOCOL parameter to be received by a
-client prior to its use. An endpoint that supports receiving the extended
-CONNECT method SHOULD send this setting with a value of 1.
+Clients negotiate the use of WebTransport ove HTTP/2 using both the SETTINGS
+frame and one or more extended CONNECT requests as defined in {{!RFC8441}}.
+
+Use of the extended CONNECT method extension requires the
+SETTINGS_ENABLE_CONNECT_PROTOCOL parameter to be received by a client prior to
+its use. An endpoint that supports receiving the extended CONNECT method SHOULD
+send this setting with a value of 1.
 
 The extended CONNECT method extension uses the `:protocol` psuedo-header field
 to negotiate the protocol that will be used on a given stream in an HTTP/2
-connection. This document registers a new token, "webtransport", in the "Hypertext Transfer
-Protocol (HTTP) Upgrade Token Registry" established by {{?RFC7230}} and located
-at <https://www.iana.org/assignments/http-upgrade-tokens/>.
+connection. This document registers a new token, "webtransport", in the
+"Hypertext Transfer Protocol (HTTP) Upgrade Token Registry" established by
+{{?RFC7230}} and located at
+<https://www.iana.org/assignments/http-upgrade-tokens/>.
 
 This token is used in the `:protocol` psuedo-header field to indicate that the
-endpoint wishes to use the WebTransport protocol on the new stream. If
-successful, the resulting stream established via the extended CONNECT handshake
-is referred to as a WebTransport Connect Stream, or simply a Connect Stream.
+endpoint wishes to use the WebTransport protocol on the new stream.
 
 
 ## The WTHEADERS Frame
 
-A new HTTP/2 frame called `WTHEADERS` is introduced for establishing streams in
-a bidirectional manner. A stream opened by a `WTHEADERS` frame is referred to as
-a WebTransport Stream, and it MAY be continued by CONTINUATION and DATA frames.
-WebTransport Streams can be initiated by either clients or servers via a
-`WTHEADERS` frame that refers to the corresponding client-initiated extended
-CONNECT stream on which the WebTransport protocol was negotiated.
+Clients or servers use a WTHEADERS frame to initate a new WebTransport Stream.
 
-The WTHEADERS frame (type=0xfb) has all the fields and frame header flags defined
-by HEADERS frame in HEADERS {{!RFC7540}}, Section 6.2.
+The WTHEADERS frame (type=0xfb) has all the fields and frame header flags
+defined by HEADERS frame in HEADERS {{!RFC7540}}, Section 6.2.
 
 The WTHEADERS frame has one extra field, Connect Stream ID. WTHEADERS frames can
 be sent on a stream in the "idle", "open", or "half-closed (remote)" state.
```

Comment:
Line 494: "reset all..." - I think this conflicts with other language that treats graceful and ungraceful termination differently.

Reply:
Let's tackle that in an issue I think we require both behaviors in different places and that can't be right :)